### PR TITLE
feat(ValuedExpress): allocating value for token flows

### PR DESCRIPTION
**Pull request by re1ro**
* [x] `AxelarValuedExpressExecutable`: allocating value before `_execute` for token flows

### CODE REVIEW

**Review by Foivos — COMMENTED** [review]
Why did you change the order of the to and from functions? It makes checking for changes quote hard

File: contracts/express/AxelarValuedExpressExecutable.sol [hunk 1]
```diff
@@ -203,31 +204,37 @@ abstract contract AxelarValuedExpressExecutable is ExpressExecutorTracker, IAxel
...
         address expressExecutor,
         address tokenAddress,
         uint256 value
-    ) internal {
+    ) internal virtual {
         if (value == 0) return;
 
         if (tokenAddress == address(0)) {
-            payable(expressExecutor).safeNativeTransfer(value);
+            if (msg.value < value) revert InsufficientValue();
```
Comment (StephenFluin) [inline contracts/express/AxelarValuedExpressExecutable.sol:221]: Should the value be exact?

Comment (re1ro) [reply contracts/express/AxelarValuedExpressExecutable.sol:221]: What if there is more value provided to make another GMP call and pay gas? I think we should allow for extra value to be passed.

**Comment by re1ro** [comment]
@Foivos yeah sorry, the only real change there is to allow to pass more value than estimated. To pay gas for consequent GMP calls etc

```
if (msg.value < value) revert InsufficientValue();
```

**Review by StephenFluin — DISMISSED** [review]

**Pull request closed without merging**